dockstore-integration-testing/src/test/java/io/dockstore/client/cli/CRUDClientIT.java

```diff
@@ -282,6 +282,25 @@ public void testRefreshingHostedTool() {
         });
     }
 
+    /**
+     * This tests that you cannot register a duplicate hosted tool
+     */
+    @Test
+    public void testHostedToolDuplicate() {
+        final ApiClient webClient = getWebClient(ADMIN_USERNAME);
+        HostedApi hostedApi = new HostedApi(webClient);
+        DockstoreTool hostedTool = hostedApi.createHostedTool("awesomeTool", "cwl", "quay.io", "coolNamespace");
```

Comment:
Was recently in the area, use the enums for this stuff
https://github.com/ga4gh/dockstore/pull/1710/files#diff-ec6e0cbc48d79e219cd04155bd31dcdfR784